backend/internal/middleware/router.go

```diff
@@ -16,6 +16,13 @@ func SetupRoutes(app *gin.Engine) {
 	// usernameとpasswordを受けとり、サインインする
 	app.POST("/signin", controllers.SignIn)
 
+	// 認証が必要なエンドポイント
+	auth := app.Group("/")
+	auth.Use(AuthRequired())
+	{
+		auth.GET("/user", controllers.AuthUser)
+	}
+
```

Comment:
実装ありがとう！！！
ツイート周りのエンドポイントにも認証必須にしたい

```suggestion
	// 認証が必要なエンドポイント
	auth := app.Group("/")
	auth.Use(AuthRequired())
	{
		auth.GET("/user", controllers.AuthUser)
		app.GET("/tweets", controllers.AllTweets) // AllTweets の関数を直接呼び出す
		app.GET("/tweets/:tweetId", controllers.TweetDetail) // tweetIdでツイートの詳細を取得
		app.POST("/tweets", controllers.PostTweetDetail)     // ツイートを投稿
	}

```

Reply:
レビューありがとう！
最新のDevをマージしてコンフリクトも解消してます。

動作確認
- cookieなし
`curl http://localhost:9000/tweets/1  `
```
{"error":"Unauthorized"}
```

- Cookieあり
`curl -X 'GET' 
  'http://localhost:9000/tweets/3' 
  -H 'accept: application/json' 
  -H 'Cookie: jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MTk2NDY3OTYsImlkIjoxMDl9.1c9ow2Jt_mjI7jXv16ghELiiH64Fz7gptFSLrGIWHU0'`

```
{"id":3,"user_id":1,"title":"今日のランチ","body":"新宿で美味しいラーメンを食べた🍜 #ランチ #ラーメン","created_at":"2024-06-28T11:15:38+09:00","updated_at":"2024-06-28T11:15:38+09:00","deleted_at":null}
```